packages/flame_3d/lib/src/parser/gltf/accessor.dart

```diff
@@ -0,0 +1,309 @@
+        'is not a multiple of the stride $step.',
+      );
+    }
+
+    for (var cursor = byteOffset;
+        cursor < bytes.lengthInBytes;
+        cursor += step) {
+      yield componentType.parseData(byteData, cursor: cursor);
+    }
+  }
+
+  List<T> _typedData<T>(int size, T Function(List<num>) producer) {
+    _verifyNotSparse();
+    _verifyAccessorType(size);
+
+    final buffer = bufferView.get();
+    final view = data().toList();
+
+    final int step;
+    final byteStride = buffer.byteStride;
+    if (byteStride != null) {
+      step = byteStride ~/ componentType.byteSize;
+    } else {
+      step = size;
+    }
+    if (step == 0) {
+      throw Exception('Step cannot be 0');
+    }
+    if (view.length % step != 0) {
+      throw Exception(
+        'Accessor data length ${view.length}'
+        ' is not a multiple of the step $step',
+      );
+    }
+
+    final result = <T>[];
+    for (var i = 0; i < view.length; i += step) {
+      result.add(producer(view.sublist(i, i + size)));
+      if (result.length == count) {
+        break;
+      }
+    }
+    return result;
+  }
+
+  void _verifyAccessorType(int size) {
+    if (type.size != size) {
+      throw Exception('Accessor type mismatch: $type != $size');
+    }
+  }
+
+  void _verifyNotSparse() {
+    if (sparse != null) {
+      throw Exception('Accessor is sparse: not supported yet.');
+    }
+  }
+
+  IntAccessor asInt() => IntAccessor(root: root, accessor: this);
+  FloatAccessor asFloat() => FloatAccessor(root: root, accessor: this);
+  Vector2Accessor asVector2() => Vector2Accessor(root: root, accessor: this);
+  Vector3Accessor asVector3() => Vector3Accessor(root: root, accessor: this);
+  Vector4Accessor asVector4() => Vector4Accessor(root: root, accessor: this);
+  QuaternionAccessor asQuaternion() => QuaternionAccessor(
+        root: root,
+        accessor: this,
+      );
+  Matrix4Accessor asMatrix4() => Matrix4Accessor(root: root, accessor: this);
+
+  RawAccessor({
+    required super.root,
+    required this.bufferView,
+    required this.byteOffset,
+    required this.componentType,
+    required this.normalized,
+    required this.count,
+    required this.type,
+    required this.max,
+    required this.min,
+    required this.sparse,
+  });
+
+  RawAccessor.parse(
+    GltfRoot root,
+    Map<String, Object?> map,
+  ) : this(
+          root: root,
+          bufferView: Parser.ref(root, map, 'bufferView')!,
+          byteOffset: Parser.integer(map, 'byteOffset') ?? 0,
+          componentType: ComponentType.parse(map, 'componentType')!,
+          normalized: Parser.boolean(map, 'normalized') ?? false,
+          count: Parser.integer(map, 'count')!,
+          type: AccessorType.parse(map, 'type')!,
+          max: Parser.floatList(map, 'max'),
+          min: Parser.floatList(map, 'min'),
+          sparse: Parser.object(root, map, 'sparse', SparseAccessor.parse),
+        );
+}
+
+abstract class TypedAccessor<T> extends GltfNode {
```

Comment:
Can this be sealed?

Reply:
it shouldn't, but you never know, I don't like limiting users for no reason (eg remember all the classes on flutter gpu we couldn't extend). but what kind of optimizations could the compiler do here, I am really unaware of this